### DISCUSSION TITLE
OneDrive continuously writing to disk

### DISCUSSION THREAD

**Discussion by JoaquimCarneiro** [abraunegg/onedrive#2003] [Q&A] [answered]
First of all thanks for the application, it easy to set up and it works almost out of the box.
I don't if this is expect behavior but Onedrive is continuously writing to disk at around 3-9MiB/s, independently of activity in the onedrive folder, with writes of ~40GB in a 16 hour day. Is this normal?
I'm using Archlinux, the onedrive-abraunegg AUR package and defaults as my configuration

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
Your claims are *very* odd and certainly this sort of thing has not been reported before.

Unless the you are *writing* data / downloading data - nothing should be written to the configured onedrive folder .... so this claim is just odd.

Please review https://github.com/abraunegg/onedrive/wiki/Generate-system-performance-data-for-performance-related-issues and generate all required items including verbose debug log

**Comment by JoaquimCarneiro** [comment]
i'm following the link to debug, but since it will take some time to do it all here is a screenshot of gnome system monitor after rebuilding the client with support for debug symbols, while running onedrive for about 10 minutes (systemctl --user start onedrive.service)
![onedrive](https://user-images.githubusercontent.com/71929153/173187808-7ac0fa05-70a6-420e-b80a-19ecba64d8e3.jpg)

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
Do you have any file indexer running?

In the past Dolphin File Indexer constantly accessing files and updating the modified timestamp. The process at fault was baloo_file_extractor. Every time this process ran, this created the issue - leading to writes.

Can you check this as well.

**Comment by JoaquimCarneiro** [comment]
I'm using gnome, so indexing is built in to the environment but limited to XDG directories and non recursively in user directory, as stated in  https://gnome.pages.gitlab.gnome.org/tracker/
I followed the https://github.com/abraunegg/onedrive/wiki/Generate-system-performance-data-for-performance-related-issues page and generated everything asked, except the performance profile because i couldn't retrieve any "readable" output. my perf command opens a application like top that is constantly changing:
[onedrive log.txt](https://github.com/abraunegg/onedrive/files/8884163/onedrive.log.txt)
i also tried generating debug logs but the standalone mode (onedrive --synchronize --verbose --verbose > debug_output.log 2>&1) generates a 24MB file and the monitor mode version, i stopped it when it reached 100MB
again here's a screenshot with cpu time info (sory for the Portuguese version of it)
![Captura de ecrã de 2022-06-11 15-24-34](https://user-images.githubusercontent.com/71929153/173193104-0c774671-9f2d-4b87-8aec-dd3f729eed79.png)

sory for my bad english

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
Please can you open a new issue, filling in all the details fully, and add your debuging there so I can start to replicate your environment and try and start to investigate this fully.

I 100% suspect that there is something going on with your system and this is not an issue with this client.

**Reply by JoaquimCarneiro** [reply]
yeah probably... i would just like to know where is the problem. I was trying to avoid opening an issue since i also suspect my system, since i can't find anyone with this problem.
i'll try opening a new issue, but probably only on Monday or after, since i'm going out and i'm only getting back on Monday or later...

**Comment by JoaquimCarneiro** [comment]
well... I tied to submit the bug report but it won't let me, because my log files are too big. I think i'm going to drop the issue and move back to rclone. I prefer this application because, instead of mounting a network drive, it downloads the files locally just like the nextcloud client...
I should studding/working but i'm going to try, as a last recourse, to do clean install from github, since i noticed a difference after rebuilding it to enable the debug flags. Also can this be because of the huge amount of small files that i have in onedrive?

**Reply by abraunegg** [reply]
Your file and file size have no impact.

What is your onedrive configuration - `onedrive --display-config`

**Reply by JoaquimCarneiro** [reply]
it's the default:
```
onedrive version                             = v2.4.18-17-gc7eabab
Config path                                  = /home/quimkaos/.config/onedrive
Config file found in config path             = false
Config option 'sync_dir'                     = /home/quimkaos/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     = 
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 5
Config option 'monitor_fullscan_frequency'   = 12
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = 
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   = 
Config option 'force_http_2'                 = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = false
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```
---
I just rebuilt from github and it writes 2.2GB when i start onedrive, using systemctl start, utill it goes to sleep mode

**Reply by abraunegg** [reply]
> well... I tied to submit the bug report but it won't let me, because my log files are too big.

did you archive the log files to .tar.gz or similar ?

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
What are your logs indicating when you run just `onedrive --monitor --verbose` ?

**Reply by JoaquimCarneiro** [reply]
```
Using 'user' Config Dir: /home/quimkaos/.config/onedrive
Using 'system' Config Dir: /etc/onedrive
No user or system config file found, using application defaults
Checking Application Version ...
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
Opening the item database ...
All operations will be performed in: /home/quimkaos/OneDrive
Application version: v2.4.18-17-gc7eabab
Account Type: business
Default Drive ID: b!SIdu3mErEEK9ZJuHECcbfVJgPA-Zoy1Fp5ezSjQi8-1qPQsyz0hrSb1fra3-3mJA
Default Root ID: 01V6UNHKV6Y2GOVW7725BZO354PWSELRRZ
Remaining Free Space: 2150321810556
Fetching details for OneDrive Root
OneDrive Root exists in the database
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Applying changes of Path ID: 01V6UNHKV6Y2GOVW7725BZO354PWSELRRZ
Updated Remaining Free Space: 2150321810556
Processing 292 OneDrive items to ensure consistent local state
Processing 319 OneDrive items to ensure consistent local state
Processing 277 OneDrive items to ensure consistent local state
Processing 302 OneDrive items to ensure consistent local state
Processing 233 OneDrive items to ensure consistent local state
Processing 269 OneDrive items to ensure consistent local state
Processing 249 OneDrive items to ensure consistent local state
Processing 263 OneDrive items to ensure consistent local state
Processing 257 OneDrive items to ensure consistent local state
Processing 357 OneDrive items to ensure consistent local state
Processing 360 OneDrive items to ensure consistent local state
Processing 307 OneDrive items to ensure consistent local state
Processing 318 OneDrive items to ensure consistent local state
Processing 333 OneDrive items to ensure consistent local state
Processing 313 OneDrive items to ensure consistent local state
Processing 265 OneDrive items to ensure consistent local state
Processing 223 OneDrive items to ensure consistent local state
Processing 258 OneDrive items to ensure consistent local state
Processing 247 OneDrive items to ensure consistent local state
Processing 250 OneDrive items to ensure consistent local state
Processing 222 OneDrive items to ensure consistent local state
Processing 222 OneDrive items to ensure consistent local state
Processing 220 OneDrive items to ensure consistent local state
Processing 239 OneDrive items to ensure consistent local state
Processing 237 OneDrive items to ensure consistent local state
Processing 281 OneDrive items to ensure consistent local state
Processing 307 OneDrive items to ensure consistent local state
Processing 325 OneDrive items to ensure consistent local state
Processing 320 OneDrive items to ensure consistent local state
Processing 303 OneDrive items to ensure consistent local state
Processing 291 OneDrive items to ensure consistent local state
Processing 291 OneDrive items to ensure consistent local state
Processing 264 OneDrive items to ensure consistent local state
Processing 266 OneDrive items to ensure consistent local state
Processing 268 OneDrive items to ensure consistent local state
Processing 286 OneDrive items to ensure consistent local state
Processing 233 OneDrive items to ensure consistent local state
Processing 220 OneDrive items to ensure consistent local state
Processing 238 OneDrive items to ensure consistent local state
Processing 230 OneDrive items to ensure consistent local state
Processing 264 OneDrive items to ensure consistent local state
Processing 206 OneDrive items to ensure consistent local state
Processing 234 OneDrive items to ensure consistent local state
Processing 235 OneDrive items to ensure consistent local state
Processing 251 OneDrive items to ensure consistent local state
Processing 236 OneDrive items to ensure consistent local state
Processing 232 OneDrive items to ensure consistent local state
Processing 254 OneDrive items to ensure consistent local state
Processing 240 OneDrive items to ensure consistent local state
Processing 231 OneDrive items to ensure consistent local state
Processing 241 OneDrive items to ensure consistent local state
Processing 234 OneDrive items to ensure consistent local state
Processing 268 OneDrive items to ensure consistent local state
Processing 211 OneDrive items to ensure consistent local state
Processing 234 OneDrive items to ensure consistent local state
Processing 235 OneDrive items to ensure consistent local state
Processing 259 OneDrive items to ensure consistent local state
Processing 204 OneDrive items to ensure consistent local state
Processing 235 OneDrive items to ensure consistent local state
Processing 234 OneDrive items to ensure consistent local state
Processing 238 OneDrive items to ensure consistent local state
Uploading differences of ~/OneDrive
Processing .
The directory has not changed
Processing Ficheiros de Conversa do Microsoft Teams
The directory has not changed
Processing Ficheiros de Conversa do Microsoft Teams/Contrato de formação_JoaquimCarneiro.pdf
The file has not changed
Processing Ficheiros de Conversa do Microsoft Teams/FAD_Avaliação feita por formandos_joaquimCarneiro.xlsx
The file has not changed
Processing Ficheiros de Conversa do Microsoft Teams/placa_de_rede.pptx
The file has not changed

[...] 5MB worth of "The file has not changed"

Processing Imagens/Feedback/{1E6F7841-6511-435B-9392-26FE9035DFDF}/Capture001.png
The file has not changed
Processing Anexos
The directory has not changed
Processing .Trash-1000
The directory has not changed
Processing .Trash-1000/info
The directory has not changed
Processing .Trash-1000/files
The directory has not changed
Processing Texto Simples.txt
The file has not changed
Uploading new items of ~/OneDrive
Applying changes of Path ID: 01V6UNHKV6Y2GOVW7725BZO354PWSELRRZ
Updated Remaining Free Space: 2150321810556
```

**Reply by abraunegg** [reply]
@JoaquimCarneiro 
That is 100% normal for application startup - it is performing the integrity check of your system. Post the initial check, when it says the sync with OneDrive is complete, the client *should* be sleeping for 300 seconds, and there *should* be zero activity to your disk

**Reply by JoaquimCarneiro** [reply]
well now it's better after building from git. after 30 minutes it wrote 2.4GB (2.2GB on startup / 0.2 after) before it would be around 4GB

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
> I think i'm going to drop the issue and move back to rclone.

rclone is a 'one-way' client .. does not fully support all the OneDrive API / OneDrive account types / SharePoint & does not cater for all the OneDrive SharePoint modifications that occur.

**Reply by JoaquimCarneiro** [reply]
exactly

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
When you build the client manually on your platform - did you use DMD or LDC ? Was this installed via a package or via different method?

**Reply by JoaquimCarneiro** [reply]
LDC
https://archlinux.org/packages/community/x86_64/ldc/

**Reply by JoaquimCarneiro** [reply]
i did:
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
./configure
make clean; make;
sudo make install

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
> I just rebuilt from github and it writes 2.2GB when i start onedrive, using systemctl start, utill it goes to sleep mode

~There is no way the client should be 'writing' 2.2GB to disk unless it is 'downloading' 2.2GB of data to your OneDrive folder.~

**Edit:** potential root cause now identified after exhaustive analysis

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
I have been doing some testing with a 3.7GB test data set, comprising of 602 random files (images, documents, videos) that simulates most user content, and it is the same data set I use to validate OneDrive functionality. It includes bad data that OneDrive has issue with.

In short - I cannot come even *close* to reproducing something you are seeing.

The onedrive client I am testing is `v2.4.18-17-gc7eabab` as built from source on Arch Linux:

![system_details](https://user-images.githubusercontent.com/4956234/173252376-44ad4c6e-c09f-45a6-be48-c89041367337.JPG)

The application configuration is using defaults:
```text
onedrive version                             = v2.4.18-17-gc7eabab
Config path                                  = /home/alex/.config/onedrive
Config file found in config path             = false
Config option 'sync_dir'                     = /home/alex/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     = 
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 5
Config option 'monitor_fullscan_frequency'   = 12
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = 
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   = 
Config option 'force_http_2'                 = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = false
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```

When downloading the initial data from OneDrive - this is when there is only activity being done with writing data to disk:

![Downloading Data](https://user-images.githubusercontent.com/4956234/173252419-40cf07b5-0b2a-40c0-8636-f9aad527099c.JPG)

Once the download of data is complete, the disk read|writes are greatly reduced

![download of data complete](https://user-images.githubusercontent.com/4956234/173252440-57a2db95-737b-41b4-b43a-e5244da984ea.JPG)

To help validate things when running via systemd, now that the data has been downloaded

![enable_systemd_begin](https://user-images.githubusercontent.com/4956234/173252473-8ac7bb58-e983-442d-8f24-7fea8fcb4fd8.JPG)

and this morning after leaving the systemd service run overnight

![running overnight](https://user-images.githubusercontent.com/4956234/173252487-01b16f90-062d-44d8-8741-bdcfe983e1a2.JPG)

Now .. admittedly, 602 files across 3.7GB is not a 'large data set' ... so, I am going to redo this test with another data set that I use that comprises of 100K randomly generated files

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
Using a much larger data set + comparing this exact same data set on different platforms - I think you have found a bug ... but this bug is not with 'onedrive' .... I think this is either a kernel bug, or simply a 'reporting bug' in the resources application ..... 

When the application is 'fully' in-sync, and you start up the application in any mode - it will validate the contents of OneDrive with what is stored locally:
```
Processing 203 OneDrive items to ensure consistent local state due to a full scan being requested
Processing 203 OneDrive items to ensure consistent local state due to a full scan being requested
Processing 203 OneDrive items to ensure consistent local state due to a full scan being requested
Processing 203 OneDrive items to ensure consistent local state due to a full scan being requested
Processing 203 OneDrive items to ensure consistent local state due to a full scan being requested
Processing 203 OneDrive items to ensure consistent local state due to a full scan being requested
```

Now .. when processing these items from OneDrive, we call stat() to query the local status of the data locally. All we do here application wise is *read* the file, *read* the DB via a DB query to validate and check that the data we have locally matches what we know to be true - this is all *read* operations.

In *newer* Linux kernels, this call however was *changed* - see https://dashdash.io/2/newfstatat that on some system architectures, this call has been changed to call newfstatat():

https://dashdash.io/2/newfstatat#Notes-C_library/kernel_differences

![newfstatat](https://user-images.githubusercontent.com/4956234/173462612-eb7a67ec-c007-4ddd-8074-459d42dbf48b.JPG)

Now ... when I am running this larger 100K file set against OneDrive Business on Arch Linux

![capture_3](https://user-images.githubusercontent.com/4956234/173462696-2e1d7d3e-a22a-46ac-890c-7788ef16587d.JPG)

Points of note:
* I am seeing what you are claiming .. a huge amount of disk writes ...... which is odd ...

But, based on `lsof` what are we writing to .. a significant amount of disk writes should be going somewhere ... so what is `lsof` telling me?

```text
# - The number in front of flag(s) is the file descriptor number of used by the process to associated with the file
u - File open with Read and Write permission
r - File open with Read permission
w - File open with Write permission
W - File open with Write permission and with Write Lock on entire file
mem - Memory mapped file, usually for share library
```

* The *only* file open with any write permissions is the database file .. which is to be expected, as, if any of the data here stale, it needs to be updated ...
* We use the shm & wal files for transient data before committal to the database .. so they could also be used / grow in size
* But as the 'Disk write total' is increasing ... no other file is being opened as evidenced by `lsof` and no file is certainly having any 3.x GB of writes to it ....

OK .. so we have worked out something odd is going on ... what happens on 'older' systems that are not using newfstatat() ?

**During Validation**
![CentOS](https://user-images.githubusercontent.com/4956234/173463420-0e809da2-87bb-42df-ad02-72c0dfbaa4b1.JPG)

**On Completion**
![CentOS_complete](https://user-images.githubusercontent.com/4956234/173463878-7a3b99d2-8d82-4f50-b5ec-a9779920a026.JPG)


Points of note:
* Exact same 100K dataset
* Disk reads are being seeing + disk writes

Something odd is going on here with the reporting of disk writes, and from my viewpoint the application is not writing to disk or to the database file (as that is the *only* file open with write permissions) - and there is certainly no 3GB of data being written into the database - which would see the DB file grow and/or the wal & shm files grow as well ....

Disk reads of file statistics to get last modified time, or to query file size is not a disk write. Performing a DB query is not a disk write.

~~I do not have a concrete answer for you ... but given on Arch Linux the disk reads are not even being evidenced - this 100% feels to me as a bug in how the filesystem | kernel is being monitored and what is being looked at to generate those statistics.~~

**Edit:** It appears that all changes, coming from OneDrive API, even if zero difference, were being upserted to the database - which, is where these writes were coming from .. writes were occurring, just no data change.

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
Digging into this further, I have found a possible cause in the application regarding how database inserts are being managed & handled for unchanged items and objects.

I will post a PR for you to test shortly.

In all my tests with what I am looking at - disk usage post a change I am testing is significantly reduced:

![image](https://user-images.githubusercontent.com/4956234/173490246-5de563a8-2ae0-419c-9890-b710452cb834.png)

**Comment by abraunegg** [comment, marked as answer]
@JoaquimCarneiro
Please can you test the following PR to resolve your issue of OneDrive continuously writing to disk:

```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/2004/head:pr2004
git checkout pr2004
./configure; make clean; make;
```
To run the PR, you need to run the client from the PR build directory:
```
./onedrive <any options needed>
```

To install the PR, you need to potentially remove the 'onedrive' package and then perform `sudo make install` to install the PR version. This will ensure that you only have 1 client binary|version on your system.

When running the PR, your version should be: `onedrive v2.4.18-19-g4e67f1d` or greater.

**Comment by JoaquimCarneiro** [comment]
First of all thanks for the attention and the effort.
I did the PR and the difference is abysmal. I made a 35 minute run and the only thing i noticed is that it re-wrote the database every 5 minutes (300 seconds), making incremental writes with that interval. here are some images:
first i did a sync
![Captura de ecrã de 2022-06-14 19-22-25](https://user-images.githubusercontent.com/71929153/173668309-d8cf8877-09ed-43e6-9ee1-c472597f232b.png)
5 minutes in
![Captura de ecrã de 2022-06-14 19-24-29](https://user-images.githubusercontent.com/71929153/173668356-dd546a17-a29c-4582-9d4a-21c8179bd684.png)
10 minutes in
![Captura de ecrã de 2022-06-14 19-29-33](https://user-images.githubusercontent.com/71929153/173668400-f98e839c-9eab-46ec-820c-c09892293dd9.png)
Finished sync (25 minutes)
![Captura de ecrã de 2022-06-14 19-44-48](https://user-images.githubusercontent.com/71929153/173668426-85c6c31d-692e-40e4-8842-11beeea22c2d.png)
Database files & sizes (35 minutes)
![Captura de ecrã de 2022-06-14 19-54-40](https://user-images.githubusercontent.com/71929153/173668454-09ff696b-23db-49a2-8457-a096050578c2.png)

all this was done without any activity with onedrive folder or service
p.s. i forgot and did a non verbose run...

**Reply by abraunegg** [reply]
@JoaquimCarneiro 
Looking at your data - what you see is not *abysmal* .... you are having *significantly* less data being constantly written.

Your items.sqlite file is in a size that is *normal* for a OneDrive content over 50K files .. this is mine using the 100K file test data:
```
[alex@arch-linux-vm-2022 onedrive]$ ls -lah
total 137M
drwx------  2 alex alex 4.0K Jun 15 05:45 .
drwx------ 14 alex alex 4.0K Jun 14 11:34 ..
-rw-r--r--  1 alex alex  67M Jun 15 05:44 items.sqlite3
-rw-r--r--  1 alex alex 160K Jun 15 05:44 items.sqlite3-shm
-rw-r--r--  1 alex alex  71M Jun 15 05:44 items.sqlite3-wal
-rw-------  1 alex alex  881 Jun 15 05:33 refresh_token
[alex@arch-linux-vm-2022 onedrive]$ 
```

> I made a 35 minute run and the only thing i noticed is that it re-wrote the database every 5 minutes (300 seconds), making incremental writes with that interval

Now - at the end of each sync, the database flushes to disk all of the 'in-memory' change | working objects - this is why the files get updated.

If any new files are downloaded (which you do) or remote folder change occurs with a change in content or last modified timestamp - the database records need to be updated .. this is why you are seeing this.

As compared to running the v2.4.18 release version - your situation should be greatly improved ....

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
Below is my overnight run using PR2004 and 100K data set, with zero change local or remote:

**Arch Linux**
![Arch_Linux_PR2004_overnight](https://user-images.githubusercontent.com/4956234/173677248-8f2b809c-71d3-40a2-bc91-2b768cad1d13.JPG)


**CentOS**
![CentOS_PR2004_overnight](https://user-images.githubusercontent.com/4956234/173677261-bd92a4fa-8ab3-4081-85bc-771249ffeb31.JPG)

The interesting aspect here is that there are *no* disk reads occurring supposedly for the process ... however .. I know for a fact that there are reads occurring for Arch Linux:

![image](https://user-images.githubusercontent.com/4956234/173677732-77bc78e9-f54a-48c5-b1d7-50fd6816993d.png)

And for CentOS
![image](https://user-images.githubusercontent.com/4956234/173677857-7973819f-138c-4f2b-9b58-195df50218d2.png)

So .. I still think that there is a *reporting* issue bug due to `newfstatat()` being used ..

The question I have right now .. why is there still some significant writes going on .. need to dig into that a little further

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
Please can you rebuild your PR client version by pulling in the latest change.

When running the application, your version should be `onedrive v2.4.18-21-g14bd247`

This gives the following performance reference on my 100K data set:

![Arch_Linux_Latest_PR_Update](https://user-images.githubusercontent.com/4956234/173685269-b3d94715-4af3-4d2a-94b4-faded33e03bf.JPG)

With the disk performance as:

![Arch_Linux_Latest_PR_Disk_Perf](https://user-images.githubusercontent.com/4956234/173685340-15378b9c-dc94-4a64-86ac-1c4efcf767da.JPG)

Please can you update your PR client version and re-test.

**Reply by JoaquimCarneiro** [reply]
To me it looks fixed. My perception is, the application starts, writes to the sqlite, shm and wal file, committing ~65MB and them do reallly small increments to shm and wal files. with 50 minutes of testing it's still at 65MB:
Started @ 22:07
![Captura de ecrã de 2022-06-14 22-07-40](https://user-images.githubusercontent.com/71929153/173695336-b32a3fa3-dc8c-423d-b5b9-15729aac430b.png)
7 minutes - sleep mode
![Captura de ecrã de 2022-06-14 22-14-21](https://user-images.githubusercontent.com/71929153/173695363-6e95888c-3947-4527-ba44-f1c510a5b2ec.png)
Finished sync
![Captura de ecrã de 2022-06-14 22-33-57](https://user-images.githubusercontent.com/71929153/173695386-7c75a61f-961b-4b32-bcf9-eea016f7c100.png)
50 minutes
![Captura de ecrã de 2022-06-14 22-57-03](https://user-images.githubusercontent.com/71929153/173696200-2e163384-f146-494c-9c69-d79a8f51f311.png)

almost no hard drive footprint, less CPU time

**Reply by abraunegg** [reply]
@JoaquimCarneiro 
> To me it looks fixed.

Correct .. was about to post the following:

**Arch Linux - Updated PR post 1Hr working:**

![Arch_Linux_Latest_PR2004_1_Hr](https://user-images.githubusercontent.com/4956234/173697006-200d17ba-2f08-420a-864e-2ad077e4be1b.JPG)

**Disk Utilisation:**

![Arch_Linux_Latest_PR2004_1_Hr_Disk_Stats](https://user-images.githubusercontent.com/4956234/173697053-6f1088f2-5bd7-4fb7-abf2-6fe305299048.JPG)

The only disk spike is during the integrity check which is 100% to be expected.

**Comment by abraunegg** [comment]
@JoaquimCarneiro 
Will merge this PR into 'master' later today

**Reply by JoaquimCarneiro** [reply]
thanks!
